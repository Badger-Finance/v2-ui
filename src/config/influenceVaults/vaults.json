{
	"rewardFrequencies": [
		{
			"name": "bveCVX, BADGER",
			"value": "Each bi-weekly bribe sale"
		},
		{
			"name": "bcvxCRV",
			"value": "Each ~2hr cycle"
		}
	],
	"rewardFrequenciesModalInfo": {
		"title": "Reward Frequency",
		"body": [
			"bveCVX and BADGER rewards are distributed at the completion of each bi-weekly voting/bribing round, after bribes for that round have been sold. Convex locking rewards are distributed each ~2hr cycle as autocompounding ",
			"[bcvxCRV](convex-cvxcrv)",
			"."
		],
		"pointType": "list",
		"points": [
			["- 75% of bribe sales (after fees): sold for bveCVX"],
			["- 25% of bribe sales (after fees): sold for BADGER"],
			["- Convex vlCVX locking rewards: claimable as autocompounding bCvxCRV"],
			["- All rewards are claimable through the app"]
		]
	},
	"withdrawModalInfo": {
		"title": "CVX Withdrawable",
		"body": [
			"Each week, this vault locks batches of CVX tokens for a 16 week period. As vlCVX unlocks on a rolling basis, CVX becomes available to withdraw from the vault. Alternatively, bveCVX may be traded for CVX via the ",
			"[Curve pool](https://curve.fi/factory/52/)",
			" at any time."
		],
		"pointType": "list",
		"points": [
			["- CVX tokens are locked each Thursday just before 00:00 UTC"],
			["- Unlocked CVX is withdrawable from 00:01 UTC each Thursday until the next locking event"],
			[
				"- The unlocking schedule for bveCVX can be found on this ",
				"[Dune dashboard](https://dune.com/tianqi/Convex-Locked-CVX-V2(Sponsored-by-Badger))",
				"."
			]
		]
	},
	"perfomanceInfo": {
		"body": [
			[
				"This vault locks 100% of deposited Convex tokens for rolling periods of 16 weeks. Badger will use vlCVX to vote for bribes during each voting round, sell them, and emit the proceeds back to holders in the form of bveCVX (autocompounded), and claimable BADGER and bcvxCRV."
			],
			[
				"Unlike other Badger Vaults, bveCVX limits the times when users may withdraw their funds. Limited pre-unlock liquidity is available through this ",
				"[Curve Pool](https://curve.fi/factory/52/)",
				". Please carefully read the ",
				"[User Guide](https://docs.badger.com/badger-finance/vaults/vault-user-guides-ethereum/vote-locked-cvx)",
				" for more information. Details on the timing of CVX unlocks are available on this ",
				"[Dune dashboard](https://dune.com/tianqi/Convex-Locked-CVX-V2(Sponsored-by-Badger))",
				"."
			]
		],
		"liquity": [["% CVX Received from 10k "], ["[bveCVX swap](https://curve.fi/factory/52/)"]]
	},
	"feeInfo": {
		"fees": [
			{
				"bveCVX Liquidity Support": "5%",
				"BADGER Liquidity Support": "5%",
				"DAO Operations Fee": "5%"
			}
		],
		"feeModalInfo": {}
	}
}
