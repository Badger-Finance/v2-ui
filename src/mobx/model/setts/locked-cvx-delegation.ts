export enum DelegationState {
	Eligible = 'eligible',
	Ineligible = 'ineligible',
	Delegated = 'delegated',
	BadgerDelegated = 'badger-delegated',
}
