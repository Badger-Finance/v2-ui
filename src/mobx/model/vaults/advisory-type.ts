export enum AdvisoryType {
	AuraLock = 'VaultLock',
	VaultLock = 'VaultLock',
	ConvexLock = 'ConvexLock',
	Remuneration = 'Remuneration',
	Chadger = 'Chadger',
	None = 'None',
}
