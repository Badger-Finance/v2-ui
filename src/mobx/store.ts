import { RouterStore } from 'mobx-router';
import UiState from './reducers';
import WalletStore from './stores/walletStore';
import ContractsStore from './stores/contractsStore';
import AirdropStore from './stores/airdropStore';
import RebaseStore from './stores/rebaseStore';
import RewardsStore from './stores/rewardsStore';
import IbBTCStore from './stores/ibBTCStore';
import TransactionsStore from './stores/transactionsStore';
import SettStoreV2 from './stores/settStoreV2';
import { NETWORK_LIST } from '../config/constants';
import HoneyPotStore from './stores/honeyPotStore';

export class RootStore {
	public router: RouterStore<RootStore>;
	public wallet: WalletStore;
	public uiState: UiState;
	public contracts: ContractsStore;
	public airdrops: AirdropStore;
	public rebase: RebaseStore;
	public rewards: RewardsStore;
	public ibBTCStore: IbBTCStore;
	public setts: SettStoreV2;
	public transactions: TransactionsStore;
	public honeyPot: HoneyPotStore;

	constructor() {
		this.router = new RouterStore<RootStore>(this);
		this.wallet = new WalletStore(this);
		this.contracts = new ContractsStore(this);
		this.airdrops = new AirdropStore(this);
		this.rebase = new RebaseStore(this);
		this.rewards = new RewardsStore(this);
		this.uiState = new UiState(this);
		this.ibBTCStore = new IbBTCStore(this);
		this.honeyPot = new HoneyPotStore(this);
		// RenVM transactions store.
		this.transactions = new TransactionsStore(this);
		this.setts = new SettStoreV2(this);

		this.walletRefresh();
	}

	async walletRefresh(): Promise<void> {
		const chain = this.wallet.network.name;
		const refreshData = [
			this.setts.loadAssets(chain),
			this.setts.loadPrices(chain),
			this.wallet.getGasPrice(),
			this.contracts.updateProvider(),
		];
		if (chain === NETWORK_LIST.ETH) {
			refreshData.push(this.setts.loadGeysers(chain));
			refreshData.push(this.rebase.fetchRebaseStats());
		} else {
			refreshData.push(this.setts.loadSetts(chain));
		}
		await Promise.all(refreshData);

		if (this.wallet.connectedAddress) {
			this.contracts.updateProvider();
			await this.wallet.getGasPrice();
			await this.contracts.fetchContracts();
			if (chain === NETWORK_LIST.ETH) {
				this.uiState.reduceRebase();
				this.ibBTCStore.init();
				this.rewards.fetchSettRewards();
				this.uiState.reduceTreeRewards();
				this.airdrops.fetchAirdrops();
				this.uiState.reduceAirdrops();
			}
			this.uiState.reduceStats();
		}
	}
}

const store = new RootStore();

export default store;
