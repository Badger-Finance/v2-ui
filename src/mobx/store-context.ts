import { RootStore } from './store';
import { createContext } from 'react';

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;
