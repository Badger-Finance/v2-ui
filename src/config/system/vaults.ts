import { AbiItem } from 'web3-utils';

import BadgerVault from './abis/Sett.json';
import YearnWrapper from './abis/YearnWrapper.json';
import BadgerSushiVault from './abis/SushiSett.json';
import DiggVault from './abis/DiggSett.json';
import deploy from '../deployments/mainnet.json';
import bscDeploy from '../deployments/bsc.json';
import { VaultNetworkConfig } from 'mobx/model';
import { NETWORK_LIST } from '../../config/constants';

export const getVaults = (network?: string | null): VaultNetworkConfig => {
	switch (network) {
		case NETWORK_LIST.BSC:
			return {
				pancakeswap: {
					abi: BadgerVault.abi as AbiItem[],
					underlying: 'token',
					contracts: [
						bscDeploy.sett_system.vaults['native.pancakeBnbBtcb'],
						bscDeploy.sett_system.vaults['native.bBadgerBtcb'],
						bscDeploy.sett_system.vaults['native.bDiggBtcb'],
					],
					fillers: {
						symbol: ['bnbBtcb', 'bBadgerBtcb', 'bDiggBtcb'],
						isFeatured: [false, false, false],
						position: [1, 2, 3],
						isSuperSett: [false, false, false],
						withdrawAll: [true, true, true],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'getPricePerFullShare',
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'token',
						},
						{
							name: 'totalSupply',
						},
					],
				},
				yearn: {
					abi: YearnWrapper.abi as AbiItem[],
					underlying: 'token',
					contracts: [bscDeploy.sett_system.vaults['yearn.wBtc']],
					fillers: {
						symbol: ['TEST'],
						isFeatured: [false],
						position: [1],
						isSuperSett: [false],
						// withdrawAll = false means there is no withdrawAll() function on the contract and must be handled differently
						withdrawAll: [false],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'token',
						},
						{
							name: 'totalSupply',
						},
					],
				},
			};
		case NETWORK_LIST.ETH:
			return {
				uniswap: {
					abi: BadgerVault.abi as AbiItem[],
					underlying: 'token',
					contracts: [
						deploy.sett_system.vaults['native.badger'],
						deploy.sett_system.vaults['native.renCrv'],
						deploy.sett_system.vaults['native.sbtcCrv'],
						deploy.sett_system.vaults['native.tbtcCrv'],
						deploy.sett_system.vaults['native.uniBadgerWbtc'],
						deploy.sett_system.vaults['harvest.renCrv'],
						deploy.sett_system.vaults['native.uniDiggWbtc'],
						deploy.sett_system.vaults['native.sushiibBTCwBTC'],
					],
					fillers: {
						symbol: ['badger', 'renCrv', 'sbtcCrv', 'tbtcCrv', 'uniBadgerWbtc', 'renCrv', 'uniDiggWbtc'],
						isFeatured: [false, false, false, false, true, true, true],
						position: [2, 11, 10, 9, 7, 8, 6],
						isSuperSett: [false, false, false, false, false, true, false, false, false],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'getPricePerFullShare',
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'token',
						},
						{
							name: 'totalSupply',
						},
					],
				},
				sushiswap: {
					abi: BadgerSushiVault.abi as AbiItem[],
					underlying: 'token',
					contracts: [
						deploy.sett_system.vaults['native.sushiWbtcEth'],
						deploy.sett_system.vaults['native.sushiBadgerWbtc'],
						deploy.sett_system.vaults['native.sushiDiggWbtc'],
						deploy.sett_system.vaults['native.sushiibBTCwBTC'],
					],
					fillers: {
						isFeatured: [false, true, true],
						position: [5, 4, 3, 6],
						symbolPrefix: ['sushi', 'sushi', 'sushi', 'sushi'],
						pairContract: [
							deploy.tokens['sushi.wBTC-WETH'],
							deploy.tokens['sushi.wBTC-BADGER'],
							deploy.tokens['sushi.wBTC-DIGG'],
							deploy.tokens['sushi.ibBTC-wBTC'],
						],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'getPricePerFullShare',
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'token',
						},
						{
							name: 'totalSupply',
						},
					],
					growthEndpoints: [
						'https://api.thegraph.com/subgraphs/name/sushiswap/master-chef',
						'https://apy.sushiswap.fi/xsushi',
						'https://apy.sushiswap.fi/?pairs=',
					],
				},
				digg: {
					abi: DiggVault.abi as AbiItem[],
					underlying: 'token',
					contracts: [deploy.sett_system.vaults['native.digg']],
					fillers: {
						isFeatured: [true],
						position: [1],
						symbolPrefix: [''],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'getPricePerFullShare',
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'totalSupply',
						},
						{
							name: 'token',
						},
					],
				},
				yearn: {
					abi: YearnWrapper.abi as AbiItem[],
					underlying: 'token',
					contracts: [deploy.sett_system.vaults['yearn.wBtc']],
					fillers: {
						symbol: ['byvWBTC'],
						isFeatured: [false],
						position: [1],
						isSuperSett: [false],
						// withdrawAll = false means there is no withdrawAll() function on the contract and must be handled differently
						withdrawAll: [false],
					},
					methods: [
						{
							name: 'balanceOf',
							args: ['{connectedAddress}'],
						},
						{
							name: 'balance',
						},
						{
							name: 'symbol',
						},
						{
							name: 'decimals',
						},
						{
							name: 'token',
						},
						{
							name: 'totalSupply',
						},
					],
				},
			};
		default:
			return {};
	}
};
