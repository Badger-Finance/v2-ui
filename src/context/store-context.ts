import { createContext } from "react";
import { RootStore } from "../mobx/store";

// Use React context to make your store available in your application
export const StoreContext = createContext({} as RootStore);
export const StoreProvider = StoreContext.Provider;