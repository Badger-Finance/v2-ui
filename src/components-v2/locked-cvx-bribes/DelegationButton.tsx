import { Button, makeStyles, Typography } from '@material-ui/core';
import { observer } from 'mobx-react-lite';
import React from 'react';

import { DelegationState } from '../../mobx/model/vaults/locked-cvx-delegation';
import { StoreContext } from '../../mobx/store-context';

const useStyles = makeStyles((theme) => ({
	badgerDelegatedTextContainer: {
		marginTop: theme.spacing(1),
	},
	badgerDelegatedText: {
		fontSize: 13,
		fontWeight: 400,
	},
}));

const DelegationButton = observer(() => {
	const { lockedCvxDelegation } = React.useContext(StoreContext);

	const classes = useStyles();
	const { delegationState, canUserDelegate } = lockedCvxDelegation;

	if (delegationState === DelegationState.BadgerDelegated) {
		return (
			<div>
				<Button size="small" variant="contained" color="primary" disabled>
					Delegate to Badger
				</Button>
				<div className={classes.badgerDelegatedTextContainer}>
					<Typography className={classes.badgerDelegatedText}>{"You've already delegated."}</Typography>
					<Typography className={classes.badgerDelegatedText}>{"Thanks, you're a top badger!"}</Typography>
				</div>
			</div>
		);
	}

	return (
		<Button
			size="small"
			variant="contained"
			color="primary"
			disabled={!canUserDelegate}
			onClick={() => lockedCvxDelegation.delegateLockedCVX()}
		>
			Delegate to Badger
		</Button>
	);
});

export default DelegationButton;
