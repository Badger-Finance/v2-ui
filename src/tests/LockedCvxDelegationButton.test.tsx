import { StoreProvider } from 'mobx/stores/store-context';
import React from 'react';

import DelegationButton from '../components-v2/locked-cvx-bribes/DelegationButton';
import { DelegationState } from '../mobx/model/vaults/locked-cvx-delegation';
import LockedCvxDelegationStore from '../mobx/stores/lockedCvxDelegationStore';
import store from '../mobx/stores/RootStore';
import { customRender } from './Utils';

describe('LockedCvxDelegationButton', () => {
	it('handles eligible state', () => {
		jest.spyOn(LockedCvxDelegationStore.prototype, 'canUserDelegate', 'get').mockReturnValue(true);
		store.lockedCvxDelegation.delegationState = DelegationState.Eligible;
		const { container } = customRender(
			<StoreProvider value={store}>
				<DelegationButton />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});

	it('handles ineligible state', () => {
		store.lockedCvxDelegation.delegationState = DelegationState.Ineligible;
		jest.spyOn(LockedCvxDelegationStore.prototype, 'canUserDelegate', 'get').mockReturnValue(false);
		const { container } = customRender(
			<StoreProvider value={store}>
				<DelegationButton />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});

	it('handles delegated state', () => {
		store.lockedCvxDelegation.delegationState = DelegationState.Delegated;
		jest.spyOn(LockedCvxDelegationStore.prototype, 'canUserDelegate', 'get').mockReturnValue(true);
		const { container } = customRender(
			<StoreProvider value={store}>
				<DelegationButton />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});

	it('handles badger delegated state', () => {
		jest.spyOn(LockedCvxDelegationStore.prototype, 'canUserDelegate', 'get').mockReturnValue(true);
		store.lockedCvxDelegation.delegationState = DelegationState.BadgerDelegated;
		const { container } = customRender(
			<StoreProvider value={store}>
				<DelegationButton />
			</StoreProvider>,
		);
		expect(container).toMatchSnapshot();
	});
});
