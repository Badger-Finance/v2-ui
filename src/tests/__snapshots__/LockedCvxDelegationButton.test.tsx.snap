// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LockedCvxDelegationButton handles badger delegated state 1`] = `
<div>
  <button
    class="MuiButtonBase-root-32 MuiButton-root-3 MuiButton-contained-11 makeStyles-badgerDelegated-1 MuiButton-containedPrimary-12 MuiButton-containedSizeSmall-22 MuiButton-sizeSmall-24 MuiButton-disabled-16 MuiButtonBase-disabled-33"
    disabled=""
    tabindex="-1"
    type="button"
  >
    <span
      class="MuiButton-label-4"
    >
      Already Delegated
    </span>
  </button>
</div>
`;

exports[`LockedCvxDelegationButton handles delegated state 1`] = `
<div>
  <button
    class="MuiButtonBase-root-32 MuiButton-root-3 MuiButton-contained-11 MuiButton-containedPrimary-12 MuiButton-containedSizeSmall-22 MuiButton-sizeSmall-24"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-4"
    >
      Delegate to Badger
    </span>
    <span
      class="MuiTouchRipple-root-35"
    />
  </button>
</div>
`;

exports[`LockedCvxDelegationButton handles eligible state 1`] = `
<div>
  <button
    class="MuiButtonBase-root-32 MuiButton-root-3 MuiButton-contained-11 MuiButton-containedPrimary-12 MuiButton-containedSizeSmall-22 MuiButton-sizeSmall-24"
    tabindex="0"
    type="button"
  >
    <span
      class="MuiButton-label-4"
    >
      Delegate to Badger
    </span>
    <span
      class="MuiTouchRipple-root-35"
    />
  </button>
</div>
`;

exports[`LockedCvxDelegationButton handles ineligible state 1`] = `
<div>
  <span
    class=""
    title="You don't have any locked CVX balance to delegate."
  >
    <button
      class="MuiButtonBase-root-43 MuiButton-root-14 MuiButton-contained-22 MuiButton-containedPrimary-23 MuiButton-containedSizeSmall-33 MuiButton-sizeSmall-35 MuiButton-disabled-27 MuiButtonBase-disabled-44"
      disabled=""
      tabindex="-1"
      type="button"
    >
      <span
        class="MuiButton-label-15"
      >
        Delegate to Badger
      </span>
    </button>
  </span>
</div>
`;
